src/vm/hints/hint_utils.rs

```diff
@@ -37,6 +37,46 @@ pub fn get_list_from_scope(vm: &mut VirtualMachine, name: &str) -> Option<Vec<Bi
     }
     val
 }
+
+pub fn get_list_from_scope_ref<'a>(
+    vm: &'a mut VirtualMachine,
+    name: &'a str,
+) -> Option<&'a Vec<BigInt>> {
+    let mut val: Option<&'a Vec<BigInt>> = None;
+    if let Some(variables) = vm.exec_scopes.get_local_variables() {
+        if let Some(PyValueType::List(py_val)) = variables.get(name) {
+            val = Some(py_val);
+        }
+    }
+    val
+}
+
+pub fn get_list_from_scope_mut<'a>(
+    vm: &'a mut VirtualMachine,
+    name: &'a str,
+) -> Option<&'a mut Vec<BigInt>> {
+    let mut val: Option<&'a mut Vec<BigInt>> = None;
+    if let Some(variables) = vm.exec_scopes.get_local_variables() {
+        if let Some(PyValueType::List(py_val)) = variables.get_mut(name) {
+            val = Some(py_val);
+        }
+    }
+    val
+}
+
+pub fn get_key_to_list_map_from_scope_mut<'a>(
+    vm: &'a mut VirtualMachine,
+    name: &'a str,
+) -> Option<&'a mut HashMap<BigInt, Vec<BigInt>>> {
+    let mut val: Option<&'a mut HashMap<BigInt, Vec<BigInt>>> = None;
+    if let Some(variables) = vm.exec_scopes.get_local_variables() {
+        if let Some(PyValueType::KeyToListMap(py_val)) = variables.get_mut(name) {
+            val = Some(py_val);
+        }
+    }
+    val
+}
```

Comment:
Returning `Result` is probably better. Trying to access an undefined symbol is an error in pretty much any language, so you're likely converting it to an error at the call site any way.

Reply:
resolved: https://github.com/lambdaclass/cleopatra_cairo/pull/319/commits/b05585e49c5fb7a3bd71b949ebfa8f4ff1f049d8